### PR TITLE
V3.6.0 batcher2d code sorting

### PR DESCRIPTION
**Pull request by Zhmz**
Re: #

related PR
https://github.com/cocos/cocos-engine/pull/11592

### Changelog

* optimize: parentOpacity move into method parameters
* optimize: handleDynamic and handleStatic integretion
* optimize: sharedBuffer divide into float and uint8 for save memory

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/2d/renderer/RenderEntity.h [hunk 1]
```diff
@@ -41,13 +41,15 @@ enum class RenderEntityType {
...
-    float colorG;
-    float colorB;
-    float colorA;
-    float colorDirtyBit;
-    float localOpacity;
-    float enabledIndex;
+    float colorR{255.0f};
+    float colorG{255.0f};
+    float colorB{255.0f};
+    float colorA{255.0f};
```
Comment (dumganhar) [inline native/cocos/2d/renderer/RenderEntity.h]: Is Color value  from 0 to 255?

Comment (dumganhar) [reply native/cocos/2d/renderer/RenderEntity.h]: Could it be uint8_t?

Comment (Zhmz) [reply native/cocos/2d/renderer/RenderEntity.h]: done

**Review by dumganhar — APPROVED** [review]

**Review by dumganhar — COMMENTED** [review]
Color(static_cast<uint8_t>(_entityAttrLayout.colorR)

static_cast is not needed now.

**Review by SantyWang — APPROVED** [review]

**Pull request merged**